source/dbo/Stored Procedures/Power BI Reporting/uspGenerateInitialSputumResults.sql

```diff
@@ -0,0 +1,79 @@
+/*
+Populate the InitialSputumSmearResult and InitialSputumPCRResult fields for each case
+Note: in all cases ‘sputum’ includes both sample types sputum - induced and sputum - spontaneous
+Calculated as follows for both smear and pcr:
+	If (any Positive or Negative results): 
+		Find results r from earliest date
+		if (r only contains negative results):
+			'Negative'
+		else:
+			'Positive'
+	Else if (any results are 'No result available'):
+		'No Result available'
+	Else if (any results 'Awaiting'):
+		'Awaiting'
+	Else (no results):
+		'No result'
```

Comment:
I'm not a fan of having pseudocode in comments, as they are usually to explain the code in words if the code doesn't explain itself (so if someone didn't understand code, they would still have trouble here)